front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/V2PipelineTemplateController.java

```diff
@@ -62,29 +74,44 @@
   }
 
   @RequestMapping(value = "", method = RequestMethod.POST)
-  void save(@RequestBody PipelineTemplate pipelineTemplate) {
-    checkForDuplicatePipelineTemplate(pipelineTemplate.getId());
-    getPipelineTemplateDAO().create(pipelineTemplate.getId(), pipelineTemplate);
+  void save(@RequestParam(value = "version", required = false) String version, @RequestBody PipelineTemplate pipelineTemplate) {
+    String templateId = StringUtils.isNotEmpty(version) ? String.format("%s:%s", pipelineTemplate.getId(), version) : pipelineTemplate.getId();
+    checkForDuplicatePipelineTemplate(templateId);
+    getPipelineTemplateDAO().create(templateId, pipelineTemplate);
+    saveDigest(pipelineTemplate);
   }
 
-  @RequestMapping(value = "{id}", method = RequestMethod.GET)
-  PipelineTemplate get(@PathVariable String id) {
-    return getPipelineTemplateDAO().findById(id);
+  private void saveDigest(PipelineTemplate pipelineTemplate) {
+    String digestId = String.format("%s@SHA256%s", pipelineTemplate.getId(), computeSHA256Digest(pipelineTemplate));
```

Comment:
I'd probably format this to have a `:` before the actual digest value, just to delineate between the hash type & the value.

Reply:
Good idea.